odl/test/solvers/functional/default_functionals_test.py

```diff
@@ -516,5 +516,75 @@ def test_moreau_envelope_l2_sq(space, sigma):
                             x * 2 / (1 + 2 * sigma))
 
 
+def test_weighted_proximal(space):
+    """Test for the weighted proximal of a SeparableSum functional."""
+
+    l1 = odl.solvers.L1Norm(space)
+    l2 = odl.solvers.L2Norm(space)
+    func = odl.solvers.SeparableSum(l1, l2)
+
+    x = func.domain.one()
+
+    sigma = [0.5, 1.0]
+
+    prox = func.proximal(sigma)(x)
+    assert all_almost_equal(prox, [l1.proximal(sigma[0])(x[0]),
+                                   l2.proximal(sigma[1])(x[1])])
+
+
+def test_weighted_proximal_L2_norm_squared(space):
+    """Test for the weighted proximal of the squared L2 norm"""
+
+    # Define the functional on the space.
+    func = odl.solvers.L2NormSquared(space)
+
+    # Set the stepsize as a random element of the spaces
+    # with elements between 1 and 10.
+    sigma = odl.phantom.uniform_noise(space, 1, 10)
+
+    # Start at the one vector.
+    x = space.one()
+
+    # Calculate the proximal point in-place and out-of-place
+    p1 = space.element()
+    func.proximal(sigma)(x, out=p1)
+    p2 = func.proximal(sigma)(x)
+
+    # Both should contain the same vector now.
+    assert all_almost_equal(p1, p2)
+
+    # Check if the subdifferential inequalities are satisfied.
+    # p = prox_{sigma * f}(x) iff (x - p)/sigma = grad f(p)
+    assert all_almost_equal(func.gradient(p1),
+                            (x - p1) / sigma)
+
+
+def test_weighted_proximal_L1_norm(space):
+    """Test for the weighted proximal of the L1 norm"""
```

Comment:
This (and the above) function only tests in the "high x domain", e.g. it only validates that the correct behaviour occurs in the linear domain of the l1-proximal. However, we would also need to know that the scaling works in the non-linear domain (basically in [-sigma, sigma]), so the test has to somehow include that. The easiest way is to compare against a manually computed result I guess.